packages/dashboard/src/components/SubwayNav/SubwayNav.tsx

```diff
@@ -0,0 +1,289 @@
+        )}
+      >
+        <Icon iconName={iconProps.iconName} className={css(iconProps.iconClassName, classNames.subwayNavStepIcon)} />
+        <Link
+          // tslint:disable-next-line jsx-no-lambda
+          onClick={() => {
+            stepToRender.onClickStep(stepToRender, undefined);
+          }}
+        >
+          <span className={css(classNames.stepLabel, stepToRender.state === SubwayNavStepState.Current ? classNames.boldStep : undefined)}>
+            {stepToRender!.label}
+          </span>
+        </Link>
+      </div>
+    );
+
+    /** Render substeps if current step has substeps */
+    if (renderSubStep && stepToRender!.subSteps) {
+      let subStepElements: JSX.Element[] = [];
+
+      stepElements.push(
+        <div
+          className={css(classNames.subwayNavStepConnector, classNames.stepConnectorCompleted)}
+          key={'connector-sub' + '-' + stepToRender.key}
+        />
+      );
+
+      const subSteps = stepToRender.subSteps;
+
+      if (subSteps !== undefined) {
+        subSteps.forEach((subStep: ISubwayNavStep, index: number) => {
+          subStepElements = subStepElements.concat(this._renderSubStep(stepToRender, subStep, index > 0 ? subSteps[index] : undefined));
+        });
+
+        stepElements = stepElements.concat(subStepElements);
+      }
+    }
+
+    return stepElements;
+  }
+
+  /**
+   * Render substeps of current step in subway Nav control
+   * @param parentStep
+   * @param subStepToRender
+   * @param prevSubStep
+   */
+  private _renderSubStep(
+    parentStep: ISubwayNavStep | undefined,
+    subStepToRender: ISubwayNavStep | undefined,
+    prevSubStep: ISubwayNavStep | undefined
+  ): JSX.Element[] {
+    const stepElements: JSX.Element[] = [];
+    const { steps } = this.props;
+    let iconProps;
+
+    if (steps !== undefined) {
+      if (parentStep !== undefined) {
+        if (subStepToRender !== undefined) {
+          // Add the connectiong line
+          if (prevSubStep !== undefined) {
+            const connectorClass = this._getStepConnectorClassName(subStepToRender);
+            stepElements.push(
+              <div className={css(classNames.subwayNavSubStepConnector, connectorClass)} key={'connector' + '-' + subStepToRender.key} />
+            );
+          }
+
+          iconProps = this._getIconPropsSubStep(subStepToRender);
+
+          stepElements.push(
+            <div
+              key={subStepToRender.key}
+              className={css(
+                classNames.subwayNavStepDiv,
+                subStepToRender.state !== SubwayNavStepState.Current && subStepToRender.disabled ? classNames.disableStep : undefined
+              )}
+            >
+              <Icon iconName={iconProps.iconName} className={css(iconProps.iconClassName, classNames.subwayNavSubStepIcon)} />
+              <Link
+                // tslint:disable-next-line jsx-no-lambda
```

Comment:
No lambdas again. This will break how React optimizes rendering.

Reply:
Will think about alternative...